### PR TITLE
EVEREST-862 Add column hide state to local storage

### PR DESCRIPTION
**Pull request by filipmikes1**
EVEREST-862

### CODE REVIEW

File: packages/ui-lib/src/table/usePersistentColumnVisibility.utils.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  const clearedValues: { [key: string]: boolean } = {};
+  for (const [key, value] of Object.entries(localStorageValue)) {
+    if (!value) {
+      clearedValues[key] = value;
+    }
+  }
+  return clearedValues;
+};
+
+export const isObjectEmpty = (obj: object) => Object.keys(obj).length > 0;
```
Comment (fabio-silva) [inline packages/ui-lib/src/table/usePersistentColumnVisibility.utils.ts:15]: if the object is empty, shouldn't length be 0?

File: packages/ui-lib/src/table/usePersistentColumnVisibility.utils.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { MRT_VisibilityState } from 'material-react-table';
+
+export const hidePositiveUnnecessaryValues = (
```
Comment (fabio-silva) [inline packages/ui-lib/src/table/usePersistentColumnVisibility.utils.ts:3]: not sure the naming is correct. Doesn't this function only return hidden columns?

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: packages/ui-lib/src/table/usePersistentColumnVisibility.utils.ts (hunk 1 above)
Comment (filipmikes1) [reply packages/ui-lib/src/table/usePersistentColumnVisibility.utils.ts]: True, logic is correctly applied but naming is wrong. Fixed.

File: packages/ui-lib/src/table/usePersistentColumnVisibility.utils.ts (hunk 2 above)
Comment (filipmikes1) [reply packages/ui-lib/src/table/usePersistentColumnVisibility.utils.ts]: I mean naming is correct, but your point of view is better. Renamed to filterHiddenColumns

**Review by fabio-silva — APPROVED** [review]

**Pull request merged**